script/send_new_webmentions

```diff
@@ -0,0 +1,86 @@
+#!/usr/bin/env ruby
+
+ENV["RAILS_ENV"] ||= "production"
+
+APP_PATH = File.expand_path('../../config/application', __FILE__)
+require File.expand_path('../../config/boot', __FILE__)
+require APP_PATH
+Rails.application.require_environment!
+
+LAST_STORY_KEY = "webmentions:last_story_id".freeze
+
+def endpoint_from_body(html)
+  doc = Nokogiri::HTML(html)
+
+  if !doc.css('[rel~="webmention"]').css('[href]').empty?
```

Comment:
Can this turn into a single `return doc.at_xpath()`? I'm not sure if it supports regexp to combine these variations or if you'd have to [define a function](https://www.rubydoc.info/github/sparklemotion/nokogiri/Nokogiri%2FXML%2FSearchable:xpath). But I'd like to see something like that, it took me a bit of staring back and forth to recognize that only the `rel` value was changing between the different branches.

Reply:
As I said in the PR body, this code is not mine, it has been extracted from [here](https://github.com/indieweb/webmention-client-ruby/blob/master/lib/webmention/endpoint/discover.rb#L30)

This code has been extensively used and tested, I don't really feel like refactoring it without an extensive test-suite.

Is it a real deal breaker for you?

Reply:
No, I can live with it. It's ugly, but at least it's not a core feature and it's at the end of the call graph.